src/main/java/com/bytebreak/animagic/texture/AnimagicSpriteBatch.java

```diff
@@ -0,0 +1,131 @@
+package com.bytebreak.animagic.texture;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.files.FileHandle;
+import com.badlogic.gdx.graphics.Camera;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.graphics.glutils.ShaderProgram;
+import com.badlogic.gdx.math.Affine2;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.math.Vector3;
+import com.sun.istack.internal.NotNull;
+
+public class AnimagicSpriteBatch extends SpriteBatch {
+
+    public final static int MAX_LIGHTS = 10;
+    private static final Color BLACK = Color.BLACK.cpy();
+
+    private Camera camera;
+
+
+    public AnimagicSpriteBatch(@NotNull Camera camera){
+        super();
+        this.setCamera(camera);
+        this.setShader(createShader());
+    }
+
+    public AnimagicSpriteBatch setCamera(@NotNull Camera camera){
+        if (camera == null) throw new RuntimeException("Cannot set the AnimagicSpriteBatch.camera with a null camera");
+        this.camera = camera;
+        return this;
+    }
+
+    private ShaderProgram createShader() {
+        FileHandle vertex = Gdx.files.internal("vertex.glsl");
+        FileHandle fragment = Gdx.files.internal("fragment.glsl");
+        ShaderProgram program = new ShaderProgram(vertex, fragment);
+        // u_proj and u_trans will not be active but SpriteBatch will still try to set them...
+        ShaderProgram.pedantic = false;
+        if (!program.isCompiled())
+            throw new RuntimeException("Could not compile Animagic shader: "
+                    + program.getLog());
+
+        // we are only using this many uniforms for testing purposes...!!
+        program.begin();
+        program.setUniformi("u_texture", 0);
+        program.setUniformi("u_normals", 1);
+        program.setUniformf("strength", 1);
+        //program.setUniformf("ambientIntensity", 0.1f);
+        program.setUniformf("ambientColor", new Vector3(1f, 1f, 1f));
+        program.setUniformf("resolution", new Vector2(Gdx.graphics.getWidth(), Gdx.graphics.getHeight()));
+        program.setUniformi("useShadow", true ? 1 : 0);
```

Comment:
Are these placeholders for something to be implemented later? Or am I misreading this? Looks like it will always set the value to 1, or is to try to be more explicit that true = 1?